service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -44,22 +44,10 @@
      */
     public abstract List<String> uris();
 
-    // hides implementation details
     public static Builder builder() {
-        return ImmutableServiceConfiguration.builder();
+        return new Builder();
     }
 
-    // hides implementation details
-    public interface Builder {
-
-        Builder apiToken(BearerToken apiToken);
-
-        Builder security(SslConfiguration security);
-
-        Builder uris(Iterable<String> uris);
-
-        Builder from(ServiceConfiguration otherConfig);
-
-        ServiceConfiguration build();
+    public static final class Builder extends ImmutableServiceConfiguration.Builder {
```

Comment:
`of` methods should be available as long as you annotate the variables with `@Value.Parameter` in the abstract class, no? We are experimenting with the new pattern atm. I am fine with more generated methods be accessible to the calling user as long as they are convenient to the users and given the benefits above.